functions/save_hours_test.ts

```diff
@@ -1,45 +1,57 @@
-import * as mf from "mock-fetch/mod.ts";
+import { stub } from "@std/testing/mock";
 import { SlackFunctionTester } from "deno-slack-sdk/mod.ts";
-import { assertEquals } from "https://deno.land/std@0.153.0/testing/asserts.ts";
+import { assertEquals } from "@std/assert";
 import SaveHoursFunction from "./save_hours.ts";
 
-// Replaces globalThis.fetch with the mocked copy
-mf.install();
-
-mf.mock("POST@/api/users.profile.get", () => {
-  return new Response(JSON.stringify({
-    ok: true,
-    profile: {
-      real_name: "Cactus Poke",
+function stubFetch() {
+  // Replaces globalThis.fetch with the mocked copy
+  return stub(
+    globalThis,
+    "fetch",
+    async (url: string | URL | Request, options?: RequestInit) => {
+      const request = url instanceof Request ? url : new Request(url, options);
+
+      assertEquals(request.method, "POST");
+
+      switch (request.url) {
+        case "https://slack.com/api/users.profile.get":
+          return new Response(JSON.stringify({
+            ok: true,
+            profile: {
+              real_name: "Cactus Poke",
+            },
+          }));
+        case "https://slack.com/api/apps.auth.external.get": {
+          const body = await request.formData();
+          if (body.get("external_token_id") === "INVALID_TOKEN_ID") {
+            return new Response(JSON.stringify({
+              ok: false,
+              error: "Invalid token",
+            }));
+          }
+          return new Response(JSON.stringify({
+            ok: true,
+            external_token: "GOOGLE_ACCESS_TOKEN",
+          }));
+        }
+        case "https://sheets.googleapis.com/v4/spreadsheets/undefined/values/A2:F2:append?valueInputOption=USER_ENTERED":
+          // Code to execute if expression === value2
+          return new Response(JSON.stringify({
+            ok: true,
+          }));
+        default:
+          throw Error(
+            `No stub found for ${request.method} ${request.url}\nHeaders: ${request.headers}`,
+          );
+      }
```

Comment:
QQ: Does the `request.method` assert above cause an error before this is reached? No blocker since testing expected paths makes sense to me!

Reply:
Every request sent by this app should be a post so I thought it made sense to have an "global" POST assertion